builtin/logical/mysql/path_role_create.go

```diff
@@ -51,9 +52,15 @@ func (b *backend) pathRoleCreateRead(
 	}
 
 	// Generate our username and password. MySQL limits user to 16 characters
-	displayName := req.DisplayName
-	if len(displayName) > 10 {
-		displayName = displayName[:10]
+	displayName := name
+	ul, ok := data.GetOk("username_length")
```

Comment:
Ignore the above comment. Remove fetching of `username_length` from parameters. Use `role.UsernameLength`.